### PR TITLE
Fix suggestion mode when typing potential lambda expression

### LINKED ISSUE #49656: Incorrect Auto Completion
When trying to type 
```cs
e.HasIndex(p => ...)
```

Auto completes to 
```cs
e.HasIndex(prop => ...)
```

I'm running VS 16.8.2. The target framework for this project is netcoreapp 3.1. The language version is 8.0.

### PR DESCRIPTION
**Pull request by genlu**
Fix #49656

### CODE REVIEW

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs [hunk 1]
```diff
@@ -491,6 +491,10 @@ private IEnumerable<TypeInferenceInfo> InferTypeInObjectCreationExpression(BaseO
                 }
 
                 return InferTypeInArgument(index, methods, argumentOpt, invocation);
+
+                static bool MethodAcceptsEmptyArgument(IMethodSymbol methodSymbol)
+                    => methodSymbol.Parameters.Length == 0 ||
+                       methodSymbol.Parameters.Length == 1 && methodSymbol.Parameters[0].IsParams;
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs:497]: please break into different statements, or parenthesize for clarity.

Comment (CyrusNajmabadi) [reply src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs:497]: what about a method where all the parameter are optional?  woudl that be a case we need to care about?  even if not, can you add a test where one of the overloads has an optional parameter?

thanks!

Comment (genlu) [reply src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs]: Good catch! Fixed

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by genlu** [comment]
Added fix to https://github.com/dotnet/roslyn/issues/49399

**Pull request merged**